sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/MethodOverloadOptionalParameter.cs

```diff
@@ -34,4 +34,25 @@ public class MethodOverloadOptionalParameter : MyBase, IMyInterface
             string b = "a" // Noncompliant {{This method signature overlaps the one defined on line 31, the default parameter value can't be used.}}
             ) {}
     }
+
+    public interface MyType { }
+
+    public interface Generics
+    {
+        void Foo<T>(string[] messages);
+        void Foo<T>(string[] messages, string delimiter = "\n"); // Noncompliant
+
+        void Foo2(string[] messages);
+        void Foo2<T>(string[] messages, string delimiter = "\n");
+
+        void Foo3<T>(T messages);
+        void Foo3<V>(V messages, string delimiter = "\n"); // Noncompliant
+
+        void Foo4<T>(IEnumerable<T> messages);
+        void Foo4<V>(IEnumerable<V> messages, string delimiter = "\n"); // Noncompliant
+
+        void Foo5<T>(IEnumerable<T> messages);
+        void Foo5<MyType>(MyType messages);

```

Comment:
What's the difference between Foo5<MyType> and Foo3<T>? MyType on Foo5 is a generic argument, not a real type, afaik, and you can delete the test case...